sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -823,24 +823,34 @@ public Mono<MsalToken> authenticateWithBrowserInteraction(TokenRequestContext re
             return Mono.error(LOGGER.logExceptionAsError(new RuntimeException(e)));
         }
 
-        if (options.isBrokerEnabled() && options.useOperatingSystemAccount()) {
-            return getPublicClientInstance(request).getValue().flatMap(pc ->
-                Mono.fromFuture(() ->
-                    acquireTokenFromPublicClientSilently(request, pc, null, false)).
-                    map(MsalToken::new));
-        } else {
-
+        // If the broker is enabled, try to get the token for the default account by passing
+        // a null account to MSAL. If that fails, show the dialog.
+
+        return getPublicClientInstance(request).getValue().flatMap(pc -> {
+            if (options.isBrokerEnabled() && options.useDefaultBrokerAccount()) {
+                return Mono.fromFuture(() ->
+                    acquireTokenFromPublicClientSilently(request, pc, null, false))
+                    .onErrorResume(e -> Mono.empty());
+            } else {
+                return Mono.empty();
+            }
+        })
+        .switchIfEmpty(Mono.defer(() -> {
             InteractiveRequestParameters.InteractiveRequestParametersBuilder builder =
                 buildInteractiveRequestParameters(request, loginHint, redirectUri);
 
             SynchronizedAccessor<PublicClientApplication> publicClient = getPublicClientInstance(request);
 
-            Mono<IAuthenticationResult> acquireToken = publicClient.getValue()
+            return publicClient.getValue()
                 .flatMap(pc -> Mono.fromFuture(() -> pc.acquireToken(builder.build())));
 
-            return acquireToken.onErrorMap(t -> new ClientAuthenticationException(
-                "Failed to acquire token with Interactive Browser Authentication.", null, t)).map(MsalToken::new);
-        }
+        }))
+        // If we're already throwing a ClientAuthenticationException we don't need to wrap it again.
+        .doOnError(t -> !(t instanceof ClientAuthenticationException),
```

Comment:
`doOnError` is a side-effect operator, I think you want `onErrorMap` here instead.

Reply:
Ah, yup. Good catch - not sure how that got swapped..